.circleci/config.yml

```diff
@@ -584,6 +589,49 @@ commands:
             cd << parameters.repo_path >>/study-builder
             mvn test-compile --batch-mode -DskipTests -Dmaven.repo.local=<< parameters.m2_repo >>
 
+  restore-ddp-cache:
+    description: Restore ddp cache
+    steps:
+      - run:
+          name: Download ddp-cache-md5 artifact
+          command: |
+            ddp_cache_md5='/home/circleci/.ddp-cache-md5.txt'
+            latest_artifact_url="https://circleci.com/api/v1.1/project/github/broadinstitute/ddp-study-server/latest/artifacts?circle-token=$CIRCLE_TOKEN"
+            md5_artifact_url=$(curl -sL "$latest_artifact_url" | jq -r '.[] | select(.node_index == 0) | select(.path == "ddp-cache-md5.txt") | .url')
+            if [[ -n "$md5_artifact_url" ]]; then
+              echo "Downloading artifact file [$md5_artifact_url] to file [$ddp_cache_md5]"
+              curl -sL "$md5_artifact_url?circle-token=$CIRCLE_TOKEN" -o "$ddp_cache_md5"
+            else
+              echo "No artifact file found, creating empty file [$ddp_cache_md5]"
+              touch "$ddp_cache_md5"
+            fi
+      - restore_cache:
+          keys:
+            - ddp-cache-{{ checksum "/home/circleci/.ddp-cache-md5.txt" }}
+            - ddp-cache-
+
+  save-ddp-cache:
+    description: Save ddp cache
+    steps:
+      - run:
+          name: Compute ddp-cache-md5
+          command: |
+            ddp_cache_dir='/home/circleci/.ddp-cache'
+            ddp_cache_md5='/home/circleci/.ddp-cache-md5.txt'
+            echo -n "" > "$ddp_cache_md5"
```

Comment:
Consider refactoring paths to shared variables. We do that elsewhere in file.

Reply:
Yeah, I wanted to do something like that. I thought it's a CircleCI feature but didn't find any docs on it. I'm just realizing now that it's a YAML feature. Let me get up to speed on yaml references and refactor.

Reply:
Switched to yaml references... but I have to say, that was quite a bit of boilerplate, especially to get a string into snippets of shell script.

Reply:
I agree! I keep going back to YAML and Circle docs looking for a better option but no dice. At least the refactor prevents changing a character in one place and breaking the script because we did not paste the same character elsewhere.